docs/app/technical-overview.md

```diff
@@ -0,0 +1,90 @@
+
+- [OpenAPI Specification][oas-docs]
+- [Connexion][connexion-home] ([source code][connexion-src])
+- [SQLAlchemy][sqlalchemy-home] ([source code][sqlalchemy-src])
+- [Alembic][alembic-home] ([source code](alembic-src))
+- [pydantic][pydantic-home] ([source code][pydantic-src])
+- [poetry](https://python-poetry.org/docs/) - Python dependency management
+
+[oas-docs]: http://spec.openapis.org/oas/v3.0.3
+[oas-swagger-docs]: https://swagger.io/docs/specification/about/
+
+[connexion-home]: https://connexion.readthedocs.io/en/latest/
+[connexion-src]: https://github.com/zalando/connexion
+
+[pydantic-home]:https://pydantic-docs.helpmanual.io/
+[pydantic-src]: https://github.com/samuelcolvin/pydantic/
+
+[sqlalchemy-home]: https://www.sqlalchemy.org/
+[sqlalchemy-src]: https://github.com/sqlalchemy/sqlalchemy
+
+[alembic-home]: https://alembic.sqlalchemy.org/en/latest/
+
+## Request operations
+
+- OpenAPI spec (`openapi.yml`) defines API interface: routes, requests, responses, etc.
+- Connexion connects routes to handlers via `operationId` property on a route
+    - Connexion will run OAS validations before reaching handler
+    - Connexion will run authentication code and set user in the request context
+    - The route handlers live in the route folder
+      [app/api/route/](/app/api/route/), with the `operationId` pointing
+      to the specific module and function
+- Handlers check if user in request context is authorized to perform the action
+  the request represents
+- Handlers use pydantic models to parse request bodies and construct response data
+- Connexion will run OAS validations on response format from handler
+
+## Authentication
+
+Authentication methods are defined in the `securitySchemes` block in
+`openapi.yaml`. A particular security scheme is enabled for a route via a
+`security` block on that route.
+
+Connexion runs the authentication before passing the request to the route
+handler. In the `api_key` security scheme, the `X-Auth` points to the
+function that is run to do the authentication.
+
+## Authorization
+n/a - Specific user authorization is not yet implemented for this API.
+
+## Running In Hosted Environments
+
+Application code gets run as tasks on [AWS Elastic Container Service (ECS)
+clusters][ecs-docs], backed by [AWS Fargate][fargate-docs]
+
+[ecs-docs]: https://docs.aws.amazon.com/AmazonECS/latest/developerguide/Welcome.html
+[fargate-docs]: https://docs.aws.amazon.com/AmazonECS/latest/developerguide/AWS_Fargate.html
+
+### ECS
+
+ECS tasks all share the same container as built by the `app` stage of the
+`Dockerfile`. This stage builds the project Python package (basically zip up
+[app/](/app/)) and installs it into the container environment.
+
+This also has the effect of making the package entrypoints as listed in the
+`[tool.poetry.scripts]` section of [pyproject.toml](/app/pyproject.toml)
+available on `PATH`, so ECS tasks besides the Paid Leave API server tend to set
+one of these as their Docker command.
```

Comment:
Some of this is not accurate for this project, specifically the Dockerfile stage, the zipping up (I think), and reference to the wrong project.

Reply:
I'll remove for now as we don't yet have ECS configured to give the correct details.